lib/plugins/rotate_asg_instances.rb

```diff
@@ -1,16 +1,50 @@
 # frozen_string_literal: true
 
 require 'aws-sdk'
-require_relative 'rotate_asg_instances/asg'
 
 module Moonshot
   module Plugins
     # Rotate ASG instances after update.
     class RotateAsgInstances
+      include ::Moonshot::DoctorHelper
+
+      def doctor(resources)
+        doctor_check_ssh(resources)
+      end
+
+      def pre_update(resources)
+        asg_ssh_class(resources).try_to_ssh
+      end
+
       def post_update(resources)
-        asg = ASG.new(resources)
-        asg.rotate_asg_instances
-        asg.teardown_outdated_instances
+        auto_scaling_group = Moonshot::ASG.new(resources, asg(resources),
+                                               physical_resource_id(resources))
+        auto_scaling_group.rotate_asg_instances
+        auto_scaling_group.teardown_outdated_instances
+      end
+
+      def asg(resources)
```

Comment:
Why not this just be asg class like before and that will handle looking up aws for physical resource id, getting the instances.

and for testing ssh we could do try_ssh(asg.first_instance)

Reply:
Moved methods back to `asg` and sending `first_instance_id` from `asg` to `ssh` class.